client/lib/signup/step-actions.js

```diff
@@ -109,7 +109,7 @@ function createSiteWithCart( callback, dependencies, {
 		if ( ! user.get() && isFreeThemePreselected ) {
 			setThemeOnSite( addToCartAndProceed, { siteSlug, themeSlugWithRepo } );
 		} else if ( user.get() && isFreeThemePreselected ) {
-			fetchSitesAndUser( siteSlug, setThemeOnSite.bind( this, addToCartAndProceed, { siteSlug, themeSlugWithRepo } ) );
+			fetchSitesAndUser( siteSlug, setThemeOnSite.bind( null, addToCartAndProceed, { siteSlug, themeSlugWithRepo } ) );
```

Comment:
What do you think of using an arrow function here?

```
fetchSitesAndUser( siteSlug, () => {
	setThemeOnSite( addToCartAndProceed, { siteSlug, themeSlugWithRepo } );
} );
```

This may be just me but I find it more readable :).